ledger/catchpointtracker.go

```diff
@@ -231,22 +237,23 @@ func (ct *catchpointTracker) finishFirstStage(ctx context.Context, dbRound basic
 	}
 
 	f := func(ctx context.Context, tx *sql.Tx) error {
+		cps := store.NewCatchpointSQLReaderWriter(tx)
```

Comment:
I'd say this requires a different abstraction like NewCatchpointSQLReaderWriter taking a handle and calling Atomic inside

Reply:
I'll rework transactions on a self contained PR after I finish moving the methods, they will require some extra abstractions and a fair bit of code rewrite.

Reply:
right, in subsequent ones